From 48b069e89c4b63bfa4c807f2011a7d0c422c4af1 Mon Sep 17 00:00:00 2001
From: Florian Bruhin <git@the-compiler.org>
Date: Tue, 26 Apr 2016 22:16:04 +0200
Subject: [PATCH] Remove @pyqtSlot for on_new_window

This worked fine with Python 3.5 but causes a circular import which is
hard to break with Python 3.4.

The original solution was to do @pyqtSlot(object), but that doesn't work
with PyQt 5.6 anymore...
---
 qutebrowser/completion/models/miscmodels.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/qutebrowser/completion/models/miscmodels.py b/qutebrowser/completion/models/miscmodels.py
index 4e44433..7b86b76 100644
--- a/qutebrowser/completion/models/miscmodels.py
+++ b/qutebrowser/completion/models/miscmodels.py
@@ -26,7 +26,6 @@
 from qutebrowser.utils import objreg, log
 from qutebrowser.commands import cmdutils
 from qutebrowser.completion.models import base
-from qutebrowser.mainwindow import mainwindow
 
 
 class CommandCompletionModel(base.BaseCompletionModel):
@@ -177,7 +176,6 @@ def __init__(self, parent=None):
         objreg.get("app").new_window.connect(self.on_new_window)
         self.rebuild()
 
-    @pyqtSlot(mainwindow.MainWindow)
     def on_new_window(self, window):
         """Add hooks to new windows."""
         window.tabbed_browser.new_tab.connect(self.on_new_tab)
